packages/navigator/src/browser/navigator-model.ts

```diff
@@ -14,20 +14,40 @@
  * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
  ********************************************************************************/
 
-import { injectable, inject } from 'inversify';
+import { injectable, inject, postConstruct } from 'inversify';
 import URI from '@theia/core/lib/common/uri';
-import { FileNode, FileTreeModel } from '@theia/filesystem/lib/browser';
+import { DirNode, FileChange, FileChangeType, FileNode, FileTreeModel } from '@theia/filesystem/lib/browser';
 import { TreeIterator, Iterators } from '@theia/core/lib/browser/tree/tree-iterator';
-import { OpenerService, open, TreeNode, ExpandableTreeNode } from '@theia/core/lib/browser';
-import { FileNavigatorTree } from './navigator-tree';
+import { OpenerService, open, TreeNode, ExpandableTreeNode, CompositeTreeNode } from '@theia/core/lib/browser';
+import { FileNavigatorTree, WorkspaceRootNode } from './navigator-tree';
 import { FileNavigatorSearch } from './navigator-search';
+import { WorkspaceService } from '@theia/workspace/lib/browser';
 
 @injectable()
 export class FileNavigatorModel extends FileTreeModel {
 
     @inject(OpenerService) protected readonly openerService: OpenerService;
     @inject(FileNavigatorTree) protected readonly tree: FileNavigatorTree;
     @inject(FileNavigatorSearch) protected readonly navigatorSearch: FileNavigatorSearch;
+    @inject(WorkspaceService) protected readonly workspaceService: WorkspaceService;
+    protected roots: string[] = [];
```

Comment:
please simplify, `roots` and `_hasMutlipleRoots` both are redundant, this information should be already in `WorkspaceService`

We should have a marker workspace root node if a tree a root of such type, we should read the rest of information from `WorkspaceService`.

Reply:
could you please explain what the `marker workspace node` is? 

is it something like a folder icon with a visible decorator like an error marker?

Also, did you mean that, we want to display the "root" that conceptually serves as the root of the multi-root workspace? (that "root" is only conceptual and does not really exist in file system).

Reply:
Hi @akosyakov After changing this back and forth, i feel that `_hasMutlipleRoots` can easily be removed, however i prefer to let `this.roots` stay. 

Conceptually it is nice to have only one source of truth, however, in reality, accessing the `roots` from `workspaceService` makes all the involved functions `async`, and i would have to "async - await" everywhere.